packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/utils/FlowableTests.kt

```diff
@@ -49,6 +49,24 @@ interface FlowableTests {
     @Test
     fun closingRealmDoesNotCancelFlows()
 
+    @Test
+    fun keyPath_topLevelProperty()
+
+    @Test
+    fun keyPath_nestedProperty()
+
+    @Test
+    fun keyPath_propertyBelowDefaultLimit()
+
+    @Test
+    fun keyPath_unknownTopLevelProperty()
```

Comment:
What is the difference between these three `unknownTopLevelProperty`, `unknownNestedProperty` and `invalidNestedProperty`? Aren't they conceptually the same?

Reply:
Maybe related to properties that actually exists but does not support it? Ex. decending into a `RealmAny` fields, etc.

Reply:
Only on the surface, but there are subtle differences. It came from when I was working on the implementation in Core.

1. At the topLevel `invalidProperty` doesn't really exist as we can listen to changes on all properties
2. At the nested level `unknownProperty` means that only the leaf node is not found, e.g. with the path `parent.child.prop` `parent.child` is a valid and `child` is a an Object/Collection with properties, only `prop` is not found on the class defined by `child`.
3. For invalidProperties, and the same path `parent.child.prop`. Here it is `child` that is invalid if it is e.g. a primitive type that doesn't have properties by itself. That also makes `prop` invalid but for a different reason than the 2nd case.

Maybe the name of the tests can be improved 🤔 

You can see the Core implementation here with the two different cases resulting in different errors:

https://github.com/realm/realm-core/pull/7087/files#diff-5e9c3010cf9e06dc7e87de9ac08da8359d523716d926dd8b44bc041e6bb49921R91

https://github.com/realm/realm-core/pull/7087/files#diff-5e9c3010cf9e06dc7e87de9ac08da8359d523716d926dd8b44bc041e6bb49921R114